src/qonlinetts.cpp

```diff
@@ -216,6 +287,26 @@ QString QOnlineTts::emotionApiCode(QOnlineTranslator::Engine engine, Emotion emo
         return emotionCode(emotion);
     }
 
-    setError(UnsupportedEmotion, tr("Selected emotion %1 is not supported for %2").arg(QMetaEnum::fromType<Emotion>().valueToKey(emotion), QMetaEnum::fromType<QOnlineTranslator::Engine>().valueToKey(engine)));
+    setError(UnsupportedEmotion, tr("Selected emotion %1 is not supported by %2").arg(QMetaEnum::fromType<Emotion>().valueToKey(emotion), QMetaEnum::fromType<QOnlineTranslator::Engine>().valueToKey(engine)));
     return {};
 }
+
+QString QOnlineTts::regionApiCode(QOnlineTranslator::Engine engine, Region region)
+{
+    if (engine == QOnlineTranslator::Google) {
+        return regionCode(region);
+    }
+
+    setError(UnsupportedRegion, tr("Selected region %1 is not supported by %2").arg(QMetaEnum::fromType<Region>().valueToKey(region), QMetaEnum::fromType<QOnlineTranslator::Engine>().valueToKey(engine)));
+    return {};
+}
```

Comment:
I would inline this functionality into `languageApiCode` because region codes are language codes for requests. Just to avoid confusion.